game/world/managers/objects/script/AIEventHandler.py

```diff
@@ -0,0 +1,97 @@
+from database.world.WorldDatabaseManager import WorldDatabaseManager
+from utils.constants.MiscCodes import CreatureAIEventTypes
+import random
+
+class AIEventHandler():
+    def __init__(self, creature):
+        self.creature = creature
+
+    def on_spawn(self):
+        events = WorldDatabaseManager.creature_ai_event_get_by_creature_id(self.creature.entry)        
+        if events:
+            for event in events:
+                if event.event_type == CreatureAIEventTypes.AI_EVENT_TYPE_CAST_SPELL_ON_SPAWN:
+                    chance_roll = random.randint(0, 100)
+                    if chance_roll <= event.event_chance:
+                        script = WorldDatabaseManager.creature_ai_script_get_by_id(event.action1_script)
+                        if script:
+                            self.creature.script_handler.enqueue_ai_script(self.creature, script)
+                    break
+    
+    def on_enter_combat(self):        
+        self.creature.script_handler.reset() # reset any scripts that were queued before combat (e.g. on spawn)
```

Comment:
Please make sure all comments follow the same structure (start with uppercase and end with dot). :pray: There are more comments that aren't following it but I'm just going to comment on this one.

Reply:
I hopefully fixed them all in the latest commit.